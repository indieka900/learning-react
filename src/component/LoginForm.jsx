import { useResizeWindow } from "../utils/hooks/useResizeWindow";

export function LoginForm() {
    //custom hook
    useResizeWindow();
    
    return (
        <form onSubmit={(e) => {
            e.preventDefault();
            const formData = new FormData(e.target)
            console.log(formData);
            const username = formData.get("username");
            const password = formData.get("password");
            console.log(`Username: ${username}. Password: ${password}`);
        }} style={{ padding: "25px" }}>
            <label htmlFor="username">Username: </label>
            <input type="text" name="username" id="username" style={{ margin: "20px" }} onChange={(e) => {
                console.log(`Username: ${e.target.value}`);
            }}
            />
            <br />
            <label htmlFor="password">password: </label>
            <input type="password" name="password" id="password" style={{ margin: "20px" }} onChange={(e) => {
                console.log(`Password: ${e.target.value}`);
            }} />
            <br />
            <button type="submit">Login</button>
        </form>
    )
}