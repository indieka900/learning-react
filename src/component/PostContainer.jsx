import { useContext } from "react";
import { PostContent } from "./PostContent";
import { UserContext } from "../utils/context/usercontext";

export function PostContainer(){
    const userContextDate = useContext(UserContext)
    return (
        <div>
            <PostContent/>
            <div>
                <h1>{userContextDate.firstname + " - " +userContextDate.lastname}</h1>
                <i>{userContextDate.email}</i>
                <h2>{userContextDate.id}</h2>
            </div>
        </div>
    )
}