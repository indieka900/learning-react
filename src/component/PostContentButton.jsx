import { useContext } from "react";
import { UserContext } from "../utils/context/usercontext";

export function PostContentButton() {
    const {setUserData} = useContext(UserContext)

    const handleOnClick = () => {
        setUserData((current) => ({...current, id: current.id + 2}))
    }

    return(
        <div>
            <h2><button onClick={handleOnClick}>Change</button></h2>
        </div>
    );
}