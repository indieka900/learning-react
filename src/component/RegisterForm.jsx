import { useState } from "react";

export function RegisterForm() {
  /*const [username, setUsername] = useState("");
  const [password, setpassword] = useState("");
  const [displayName, setdisplayName] = useState("");

  const isDisabled = !username || password.length < 8 || !displayName ? true : false;*/

  const [formField, setFormFields] = useState({
    username: "",
    password: "",
    displayName: "",
  });

  return (
    <form action="" method="post">
      <div>
        <label htmlFor="username">Username: </label>
        <input
          type="text"
          name="username"
          id="username"
          value={formField.username}
          onChange={(e) =>
            setFormFields((currentState) => ({
              ...currentState,
              username: e.target.value,
            }))
          }
        />
      </div>
      <div>
        <label htmlFor="password">Password: </label>
        <input
          type="password"
          name="password"
          id="password"
          value={formField.password}
          onChange={(e) =>
            setFormFields((currentState) => ({
              ...currentState,
              password: e.target.value,
            }))
          }
        />
      </div>
      <div>
        <label htmlFor="displayName">Display Name: </label>
        <input
          type="text"
          name="displayName"
          id="displayName"
          value={formField.displayName}
          onChange={(e) =>
            setFormFields((currentState) => ({
              ...currentState,
              displayName: e.target.value,
            }))
          }
        />
      </div>
      {/* <button type="submit" disabled={isDisabled}>Sign Up</button> */}
    </form>
  );
}
