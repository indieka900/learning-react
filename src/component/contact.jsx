/* eslint-disable react/prop-types */
export function Contact(props){
    var code = props.code ? props.code : 1234;
    // if (props.code){
    //     code = props.code
    // }
    return (
        <>
            <hr />
            <h2>Contact form</h2>
            <p>Username: {props.username}</p>
            <p>Country code: <b>{code}</b></p>
        </>
    )
   
}