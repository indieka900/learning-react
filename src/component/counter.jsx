import { useState, useEffect } from "react"

export function Counter() {
    const [counter, setCounter] = useState(0);
    //const [sync, setSync] = useState(false);
    useEffect(()=> {
        document.title = "Joseph Learning React" + counter
    }, [counter]);
    return (
        <div>
            <h1><span>Counter is: {counter}</span></h1>
            <button type="button" onClick={() => setCounter((count) => count + 1)}>Increase</button>
            <button type="button" onClick={() => setCounter((count) => count - 1)}>Decrease</button>
        </div>
    )
}