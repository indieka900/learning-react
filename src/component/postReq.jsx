import { useState } from "react";

export function PostData() {
  const [blogPostData, setBlogPostData] = useState({
    title: "",
    body: "",
  });
  return (
    <div>
      <form action="" method="post" onSubmit={(e) => {
        e.preventDefault();
        if (blogPostData.body && blogPostData.title){ 
            fetch('https://jsonplaceholder.typicode.com/posts', {
                method: 'POST',
                body: JSON.stringify({
                    userId: 12,
                    title: blogPostData.title,
                    body: blogPostData.body,
                }),
                headers: {
                    'Content-type': 'application/json; charset=UTF-8',
                }
            }).then((res) => res.json()).then((data) => {
                console.log(data);
            }).catch((e) => {
                console.log(e);
            })
        }
      }}>
        <label htmlFor="title">Title: </label>
        <input
          type="text"
          id="title"
          name="title"
          value={blogPostData.title}
          onChange={(e) => {
            setBlogPostData((current) => ({
              ...current,
              title: e.target.value,
            }));
          }}
        />
        <br />
        <label htmlFor="body">Body:</label>
        <input
          type="text"
          id="body"
          name="body"
          value={blogPostData.body}
          onChange={(e) => {
            setBlogPostData((current) => ({
              ...current,
              body: e.target.value,
            }));
          }}
        />
        <br />
        <button>Submit</button>
      </form>
    </div>
  );
}
