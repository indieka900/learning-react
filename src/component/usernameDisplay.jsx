/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';

export function UsernameDisplay( {username} ) {
    return (
        <div>
            <span>{username}</span>
        </div>
    );
}

UsernameDisplay.PropTypes = {
    username: PropTypes.string
}

UsernameDisplay.defaultProps = {
    username: "Joseph"
}