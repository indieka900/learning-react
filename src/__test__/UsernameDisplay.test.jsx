import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { UsernameDisplay } from "../component/usernameDisplay";

describe('UsernameDisplay', () => {
    it('Should render username', () => {
        const result = render(<UsernameDisplay/>);
        expect(result.container).toMatchSnapshot();
    });
});