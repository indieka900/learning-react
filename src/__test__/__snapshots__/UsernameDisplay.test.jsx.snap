// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UsernameDisplay > Should render username 1`] = `
<div>
  <div>
    <span />
  </div>
</div>
`;
